packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -0,0 +1,187 @@
+import { LineItemMap } from '../cart';
+import { Checkout, CheckoutService } from '../checkout';
+import { ShopperCurrency } from '../config';
+import { Order } from '../order';
+
+import BodlService from "./bodl-service";
+import { BodlEventsCheckout } from './bodl-window';
+
+export default class BodlEmitterService implements BodlService {
+    private _checkoutStarted = false;
+
+    constructor(
+        private checkoutService: CheckoutService,
+        private bodlEvents: BodlEventsCheckout
+    ) {
+
+    }
+
+    checkoutBegin(): void {
+        if (this._checkoutStarted) {
+            return;
+        }
+
+        const checkout = this.getCheckout();
+
+        if (!checkout) {
+            return;
+        }
+
+
+        const {
+            cart: {
+                cartAmount,
+                currency,
+                lineItems,
+                id,
+                coupons
+            },
+        } = checkout;
+
+        this.bodlEvents.emit('create_checkout_begin', {
+            id,
+            currency: currency.code,
+            cart_value: cartAmount,
+            coupon: (coupons || []).map(coupon => coupon.code.toUpperCase()).join(','),
+            lineItems: this.getProducts(lineItems, currency.code)
+        });
+
+        this._checkoutStarted = true;
+    }
+
+    orderPurchased(): void {
+        const order = this.getOrder();
+
+        if (!order) {
+            return;
+        }
```

Comment:
`const order = this.getOrderOrThrow();`

Reply:
I don't have this method in the class, what kind of implementation are you suggesting here?

Reply:
I was going by the assumption of injecting store here as compared to checkoutService